src/controller/handler/connections.ts

```diff
@@ -23,7 +23,7 @@ export class Connections implements IAgentResponseHandler {
 
         // if the cacheKey is in the cache then the agent has already accepted the request
         // when we only allow once, there is no need to continue with this message
-        if (this.cache.get<any>(cacheKey) && permissionState === AgentGovernance.PERMISSION_ONCE) {
+        if (await this.cache.get<any>(cacheKey) && permissionState === AgentGovernance.PERMISSION_ONCE) {
```

Comment:
The cache calls needed awaits, and that trickled up to converting various things to async. The issue before was that code was being executed out of sequence due to the lack of awaits